PR: https://github.com/odoo/odoo/pull/68443

From: f87a0eb763dc7aa0abe68747d59a302f32e66e8b
From: Jérémy Hennecart
Date: 2021-03-30 12:14:59

Breaking data model changes scores: del:2 + add:0, change matches:
-    _inherit = 'calendar.alarm'
-    alarm_type = fields.Selection(selection_add=[

Total Changes: 134

[REF] calendar: prepare alarm communication configuration

Purpose of this commit is to ease sending mail to attendees using template
by using a template record instead of an xml id. Indeed this allow having
flows using a configurable template instead of an hardcoded one.

Code in calendar_sms is split into main models to ease future improvements
related to SMS.

Taks ID-2191254
COM PR odoo/odoo#68443

================================= pseudo patch: =================================

--- a/addons/calendar_sms/models/__init__.py
+++ b/addons/calendar_sms/models/__init__.py
@@ -1,4 +1,6 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from . import calendar
+from . import calendar_alarm
+from . import calendar_alarm_manager
+from . import calendar_event

--- a/None
+++ b/addons/calendar_sms/models/calendar_alarm.py
@@ -0,0 +1,12 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class CalendarAlarm(models.Model):
+    _inherit = 'calendar.alarm'
+
+    alarm_type = fields.Selection(selection_add=[
+        ('sms', 'SMS Text Message')
+    ], ondelete={'sms': 'set default'})

--- a/None
+++ b/addons/calendar_sms/models/calendar_alarm_manager.py
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models
+
+
+class AlarmManager(models.AbstractModel):
+    _inherit = 'calendar.alarm_manager'
+
+    @api.model
+    def _send_reminder(self):
+        """ Cron method, overridden here to send SMS reminders as well
+        """
+        super()._send_reminder()
+        self._get_events_to_notify(ttype='sms')._do_sms_reminder()

--- a/addons/calendar_sms/models/calendar.py
+++ b/addons/calendar_sms/models/calendar_event.py
@@ -1,11 +1,7 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-import logging
-
-from odoo import api, fields, models, _
-
-_logger = logging.getLogger(__name__)
+from odoo import models, _
 
 
 class CalendarEvent(models.Model):
@@ -42,22 +38,3 @@ class CalendarEvent(models.Model):
 
     def _get_trigger_alarm_types(self):
         return super()._get_trigger_alarm_types() + ['sms']
-
-
-class CalendarAlarm(models.Model):
-    _inherit = 'calendar.alarm'
-
-    alarm_type = fields.Selection(selection_add=[
-        ('sms', 'SMS Text Message')
-    ], ondelete={'sms': 'set default'})
-
-
-class AlarmManager(models.AbstractModel):
-    _inherit = 'calendar.alarm_manager'
-
-    @api.model
-    def _send_reminder(self):
-        """ Cron method, overridden here to send SMS reminders as well
-        """
-        super()._send_reminder()
-        self._get_events_to_notify(ttype='sms')._do_sms_reminder()
